V1

Tako-San opened the pull request.

Tako-San committed:
Added getSectionAddr method

Tako-San committed:
Fix

Tako-San committed:
Made it work for simple case

Tako-San committed:
Added initializers

derzhavin3016 commented on include/elfloader/elfloader.hh:
```diff
@@ -23,6 +23,10 @@ public:
+  Addr getSectionAddr(const std::string &name) const;
+
+private:
+  ELFIO::section *getSectionPtr(const std::string &name) const;
```
```suggestion
  const ELFIO::section *getSectionPtr(const std::string &name) const;
```

derzhavin3016 commented on tools/simulator/main.cc:
```diff
@@ -1,24 +1,33 @@
   bool isInfo;
   app.add_flag("-i,--info", isInfo, "Turn on info logging level");
 
+  // input
+  fs::path input{};                                    // input
+  app.add_option("input", input, "input")->required(); // input
```
I suppose you should provide more descriptive comments (or remove them) and more informative help string to option

Tako-San committed:
[executor] Improved applyOffset + fixes

- Rewrote applyOfset using signAdd
- Implemented signAdd function which adds two unsinged values casting them to corresponding signed type
- Added constexpr to bit manipulation funcitons

derzhavin3016 committed:
Removed redundant SDword type

Tako-San committed:
Some upd
 * Added hasSection method to elfloader
 * Made return of getSectionPtr const
 * Updated Hart's ctor
 * Added exception handling to simulator's main

Tako-San committed:
Some

Tako-San committed:
Renamed member

Tako-San committed:
Added opTypeToString map generation

Tako-San committed:
Added trace to run() method
 * Added str() method to instruction class
 * Added str() method to RegFile class

Tako-San committed:
Updated regfile print way

Tako-San committed:
Upd

Tako-San committed:
Fix gcc CI failure

Tako-San committed:
CI UPD

Tako-San committed:
Fix

Tako-San committed:
Fix

Tako-San committed:
Fix

Tako-San committed:
Fix

derzhavin3016 commented on src/decoder/decoder.py:
```diff
@@ -209,12 +209,23 @@ def gen_hh(filename, yaml_dict):
+    to_write += "};\n\n"
 
-    to_write += "};\n"
+    to_write += "inline std::unordered_map<OpType, std::string> opTypeToString {\n"
```
```suggestion
    to_write += "inline std::unordered_map<OpType, std::string_view> opTypeToString {\n"
```

derzhavin3016 commented on src/decoder/decoder.py:
```diff
@@ -209,12 +209,23 @@ def gen_hh(filename, yaml_dict):
 
     to_write = COMMENT
-    to_write += "enum class OpType {\n    UNKNOWN = 0,\n"
+    to_write += "#include <string>\n"
```
```suggestion
    to_write += "#include <string_view>\n"
```

derzhavin3016 commented on include/common/inst.hh:
```diff
@@ -16,6 +20,24 @@ struct Instruction final {
+  std::string str() const {
+    std::stringstream ss{};
+
+    ss << std::setw(9) << opTypeToString[type] << " {";
```
I think you should consider using `at()` here
```suggestion
    ss << std::setw(9) << opTypeToString.at(type) << " {";
```

derzhavin3016 commented on tools/simulator/main.cc:
```diff
@@ -1,24 +1,54 @@
-  bool isInfo;
-  app.add_flag("-i,--info", isInfo, "Turn on info logging level");
+  lvl::level_enum loggingLevel{};
+  std::map<std::string, lvl::level_enum> map{
```
```suggestion
  std::map<std::string_view, lvl::level_enum> map{
```

Tako-San replied:
Doesn't work

Tako-San committed:
Update src/decoder/decoder.py

Co-authored-by: Andrey Derzhavin <derzhavin.andrej@yandex.ru>

Tako-San committed:
Update src/decoder/decoder.py

Co-authored-by: Andrey Derzhavin <derzhavin.andrej@yandex.ru>

Tako-San committed:
Update include/common/inst.hh

Co-authored-by: Andrey Derzhavin <derzhavin.andrej@yandex.ru>

derzhavin3016 reviewed (approved):
LGTMq+
